packages/shared/src/hooks/useActions.ts

```diff
@@ -17,49 +17,62 @@ interface UseActions {
   isActionsFetched: boolean;
 }
 
+interface ActionQueryData {
+  actions: Action[];
+  serverLoaded: boolean;
+}
+
 export const useActions = (): UseActions => {
   const client = useQueryClient();
   const { user } = useAuthContext();
   const actionsKey = generateQueryKey(RequestKey.Actions, user);
 
-  const { data: actions, isLoading } = useQuery(
+  const { data, isLoading } = useQuery(
     actionsKey,
     async () => {
-      const data = await getUserActions();
-      const current = client.getQueryData<Action[]>(actionsKey);
+      const serverData = await getUserActions();
+      const current = client.getQueryData<ActionQueryData>(actionsKey);
 
       if (!current || !Array.isArray(current)) {
-        return data;
+        return { actions: serverData, serverLoaded: true };
       }
 
-      const filtered = data.filter(({ type }) =>
+      const filtered = serverData.filter(({ type }) =>
         current.every((action) => action.type !== type),
       );
 
-      return [...current, ...filtered];
+      return { actions: [...current, ...filtered], serverLoaded: true };
     },
     { enabled: !!user, ...disabledRefetch },
   );
 
-  const isActionsFetched = !isLoading;
+  const actions = data?.actions;
+  const isActionsFetched = !isLoading && data?.serverLoaded;
```

Comment:
Cast to `boolean` as `!!data?.serverLoaded` because it would return `undefined` from `isActionsFetched`?

Reply:
Yep good spot!